nmagent/internal/wireserver.go

```diff
@@ -114,17 +116,25 @@ func (w *WireserverTransport) RoundTrip(inReq *http.Request) (*http.Response, er
 			// Wireserver gets angry when there's no Content-Length header, and returns
 			// a 411 status code:
 			req.ContentLength = int64(len(emptyJSONString))
-
-			// the Content-Type must also be manually set, because the net/http
-			// Content-Type detection can't be run without draining the Reader set
-			// for the Request Body. This is a good idea anyway, because the
-			// detection uses heuristic methods to figure out the Content-Type, which
-			// may not necessarily be correct for a short JSON string like this:
-			req.Header.Set(HeaderContentType, MimeJSON)
 		}
+
+		// the Content-Type must also be manually set, because the net/http
+		// Content-Type detection can't be run without draining the Reader set
+		// for the Request Body. This is a good idea anyway, because the
+		// detection uses heuristic methods to figure out the Content-Type, which
+		// may not necessarily be correct for a short JSON string like this:
+		req.Header.Set(HeaderContentType, MimeJSON)
 	}
 
 	// execute the request to the downstream transport
+	req2 := req.Clone(context.TODO())
```

Comment:
req.body is nil here..
2023/01/03 23:47:55 [1] start the goroutine for refreshing homeAz
⇨ http server started on [::]:9090
2023/01/03 23:47:55 [1] internal req headers: map[]
2023/01/03 23:47:55 [1] req2 body unread: <nil>
2023/01/03 23:47:55 [1] GetAzureCloud querying url: http://169.254.169.254/metadata/instance/compute/azEnvironment?api-version=2018-10-01&format=text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x47ac94]

goroutine 76 [running]:
io.ReadAll({0x0, 0x0})
        /usr/local/go/src/io/io.go:661 +0xd4
github.com/Azure/azure-container-networking/nmagent/internal.(*WireserverTransport).RoundTrip(0xc00037aee0, 0x1f6f8a0?)
        /usr/local/src/cns/nmagent/internal/wireserver.go:133 +0x518
net/http.send(0xc0000fa100, {0x1f6f8a0, 0xc00037aee0}, {0x1c4d360?, 0x1?, 0x0?})
        /usr/local/go/src/net/http/client.go:251 +0x5f7
net/http.(*Client).send(0xc000030ae0, 0xc0000fa100, {0xc0004c3be8?, 0x3?, 0x0?})
        /usr/local/go/src/net/http/client.go:175 +0x9b
net/http.(*Client).do(0xc000030ae0, 0xc0000fa100)
        /usr/local/go/src/net/http/client.go:715 +0x8fc
net/http.(*Client).Do(...)
        /usr/local/go/src/net/http/client.go:581
github.com/Azure/azure-container-networking/nmagent.(*Client).SupportedAPIs(0xc000030ab0, {0x1f8af20?, 0xc0004bc120?})
        /usr/local/src/cns/nmagent/client.go:177 +0x9f
github.com/Azure/azure-container-networking/cns/restserver.(*HomeAzMonitor).Populate(0xc000030b40, {0x1f8af20, 0xc0004bc120})
        /usr/local/src/cns/cns/restserver/homeazmonitor.go:96 +0x4e
github.com/Azure/azure-container-networking/cns/restserver.(*HomeAzMonitor).refresh(0xc000030b40)
        /usr/local/src/cns/cns/restserver/homeazmonitor.go:77 +0x78
created by github.com/Azure/azure-container-networking/cns/restserver.(*HomeAzMonitor).Start
        /usr/local/src/cns/cns/restserver/homeazmonitor.go:65 +0x56